### PR TITLE
Mamba

### PR DESCRIPTION
**Pull request by victoria-cherkas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,58 +23,72 @@ If you have any feature requests, feel free to raise an issue or contact us by e
...
+   
 3. Install all environments on your `$PROJECT` directory (Piz Daint) or on your `$SCRATCH` (Tsa), otherwise you risk filling up your $HOME directory. See below for instructions.
 
 </details>
 
 ### Create conda environment
 In the following instructions replace $PROJECT by $SCRATCH if using Tsa (instead of Piz Daint).
 
-Create a conda environment _psyplot_ with python[version>=3.7,<3.10] (psy-view requirement) and install requirements:
+Create a conda environment _psyplot_ with python[version>=3.7,<3.10] (psy-view requirement) and install requirements (if conda takes too long to solve, try using [mamba](https://mamba.readthedocs.io/en/latest/mamba-installation.html#mamba-install) or using the pinned environment: env/environment_pinned.yml):
```
Comment (AnnikaLau) [inline README.md:77]: This still needs to be updated if we now use mamba by default

**Comment by victoria-cherkas** [comment]
launch jenkins

File: README.md [hunk 2]
```diff
@@ -24,58 +23,72 @@ If you have any feature requests, feel free to raise an issue or contact us by e
...
 
-    conda env create --prefix $PROJECT/envs/psyplot -f env/environment.yml
+    mamba env create --prefix $PROJECT/envs/psyplot -f env/environment.yml
 
 To be able to activate your conda environment by simply using `conda activate psyplot` instead of the full path, add the following to your `.bashrc`:
```
Comment (AnnikaLau) [inline README.md:81]: Should we change to `mamba activate psyplot`?

File: README.md [hunk 3]
```diff
@@ -24,58 +23,72 @@ If you have any feature requests, feel free to raise an issue or contact us by e
...
 
     export CONDA_ENVS_PATH=$PROJECT/envs
 
 Activate environment (use "source activate" in case "conda activate" does not work):
```
Comment (AnnikaLau) [inline README.md:85]:
```suggestion
Activate environment:
```

File: README.md (hunk 2 above)
Comment (victoria-cherkas) [reply README.md]: Maybe best to change it, even though i think both work

File: testsuite/setup.sh [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
-# This setup script is used by the Jenkinsfile to setup miniconda and create a conda environment.
+# This setup script is used by the Jenkinsfile to setup minimamba and create a mamba environment.
 
-wget -O ${WORKSPACE}/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
-bash miniconda.sh -b -p $WORKSPACE/miniconda_${NODE_NAME}
-source $WORKSPACE/miniconda_${NODE_NAME}/etc/profile.d/conda.sh
+wget -O ${WORKSPACE}/mambaforge.sh "https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-$(uname)-$(uname -m).sh"
+bash mambaforge.sh -b -p $WORKSPACE/conda_${NODE_NAME}
+source $WORKSPACE/conda_${NODE_NAME}/etc/profile.d/conda.sh
+source $WORKSPACE/conda_${NODE_NAME}/etc/profile.d/mamba.sh
```
Comment (AnnikaLau) [inline testsuite/setup.sh:8]: why is both being sourced?

File: testsuite/test.sh [hunk 5]
```diff
@@ -2,7 +2,9 @@
 
 # This test script is used by the Jenkinsfile to run the iconarray tests with pytest.
 
-source $WORKSPACE/miniconda_${NODE_NAME}/etc/profile.d/conda.sh
+source $WORKSPACE/conda_${NODE_NAME}/etc/profile.d/conda.sh
+source $WORKSPACE/conda_${NODE_NAME}/etc/profile.d/mamba.sh
```
Comment (AnnikaLau) [inline testsuite/test.sh:6]: Same question as above

File: README.md (hunk 2 above)
Comment (AnnikaLau) [reply README.md]: already done!

File: testsuite/setup.sh (hunk 4 above)
Comment (victoria-cherkas) [reply testsuite/setup.sh:8]: https://github.com/conda-forge/miniforge#downloading-the-installer-as-part-of-a-ci-pipeline

It says there:

> Then you should create the path to conda and activate conda. Run this command:
> 
> source "${HOME}/conda/etc/profile.d/conda.sh"
> # For mamba support also run the following command
> source "${HOME}/conda/etc/profile.d/mamba.sh"

Think it basically enables both conda and mamba commands

**Pull request merged**